controllers/extensionRequests.js

```diff
@@ -0,0 +1,233 @@
+const extensionRequestsQuery = require("../models/extensionRequests");
+const { addLog } = require("../models/logs");
+const tasks = require("../models/tasks");
+const { getUsername } = require("../utils/users");
+const { EXTENSION_REQUEST_STATUS } = require("../constants/extensionRequests");
+
+/**
+ * Create ETA extension Request
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const createTaskExtensionRequest = async (req, res) => {
+  try {
+    const extensionBody = req.body;
+
+    if (req.userData.id !== extensionBody.assignee && !req.userData.roles?.super_user) {
+      return res.boom.forbidden("Only Super User can create an extension request for this task.");
+    }
+
+    const assigneeUsername = await getUsername(extensionBody.assignee);
+    const { taskData: task } = await tasks.fetchTask(extensionBody.taskId);
+    if (!task) {
+      return res.boom.badRequest("Task with this id or taskid doesn't exist.");
+    }
+    if (task.assignee !== assigneeUsername) {
+      return res.boom.badRequest("This task is assigned to some different user");
+    }
+    if (task.endsOn >= extensionBody.newEndsOn) {
+      return res.boom.badRequest("The value for newEndsOn should be greater than the previous ETA");
+    }
+    if (extensionBody.oldEndsOn !== task.endsOn) {
+      extensionBody.oldEndsOn = task.endsOn;
+    }
+
+    const prevExtensionRequest = await extensionRequestsQuery.fetchExtensionRequests({
+      taskId: extensionBody.taskId,
+      assignee: extensionBody.assignee,
+    });
+    if (prevExtensionRequest.length) {
+      return res.boom.forbidden("An extension request for this task already exists.");
+    }
```

Comment:
What happens in the case when a user has already created an extension request but has exceeded the `newEndsOn` time and wants to further extend the time for the task? 

Are we allowing such extension requests to happen?

Reply:
As per the description, there is a patch API for this.